lightning_examples/finetuning-scheduler/finetuning-scheduler.py

```diff
@@ -0,0 +1,565 @@
+# To specify a finetuning schedule, it's convenient to first generate the default schedule and then alter the thawed/unfrozen parameter groups associated with each finetuning phase as desired. Finetuning phases are zero-indexed and executed in ascending order.
+#
+# 1. Generate the default schedule to ``Trainer.log_dir`` with the name
+#    (``lightning_module.__class__.__name__``)_ft_schedule.yaml
+#
+# ```python
+#     from pytorch_lightning import Trainer
+#     from pytorch_lightning.callbacks.finetuning_scheduler import FinetuningScheduler
+#     trainer = Trainer(callbacks=[FinetuningScheduler(gen_ft_sched_only=True)])
+# ```
+#
+# 2. Alter the schedule as desired.
+#
+# <img src="side_by_side_yaml.svg" style="width: 100%; height: auto; max-width:800px;">
+#
+# 3. Once the finetuning schedule has been altered as desired, pass it to
+#    ``FinetuningScheduler`` to commence scheduled training:
+#
+# ```python
+# from pytorch_lightning import Trainer
+# from pytorch_lightning.callbacks.finetuning_scheduler import FinetuningScheduler
+#
+# trainer = Trainer(callbacks=[FinetuningScheduler(ft_schedule="/path/to/my/schedule/my_schedule.yaml")])
+# ```
+
+# %% [markdown]
+# ## Early-Stopping and Epoch-Driven Phase Transition Criteria
+#
+#
+# By default, ``FTSEarlyStopping`` and epoch-driven
+# transition criteria are composed. If a ``max_transition_epoch`` is specified for a given phase, the next finetuning phase will begin at that epoch unless ``FTSEarlyStopping`` criteria are met first.
+# If ``FinetuningScheduler.epoch_transitions_only`` is ``True``, ``FTSEarlyStopping`` will not be used
+# and transitions will be exclusively epoch-driven.
+#
+#
+# <div class="alert alert-info">
+#
+# **Tip:** Use of regex expressions can be convenient for specifying more complex schedules:
+#
+# <img src="emphasized_yaml.svg" style="width: 100%; height: auto; max-width:800px;">
+#
+# </div>
+#
+#
+#
+# The end-to-end example in this notebook ([Scheduled Finetuning For SuperGLUE](#superglue)) uses ``FinetuningScheduler`` in explicit mode to finetune a small foundational model on the [RTE](https://huggingface.co/datasets/viewer/?dataset=super_glue&config=rte) task of [SuperGLUE](https://super.gluebenchmark.com/).
+
+# %% [markdown]
+# ## Resuming Scheduled Finetuning Training Sessions
+#
+# Resumption of scheduled finetuning training is identical to the continuation of
+# [other training sessions](https://pytorch-lightning.readthedocs.io/en/latest/common/trainer.html) with the caveat that the provided checkpoint must have been saved by a ``FinetuningScheduler`` session.
+# ``FinetuningScheduler`` uses ``FTSCheckpoint`` (an extension of ``ModelCheckpoint``) to maintain schedule state with special metadata.
+#
+#
+# ```python
+# from pytorch_lightning import Trainer
+# from pytorch_lightning.callbacks.finetuning_scheduler import FinetuningScheduler
+# trainer = Trainer(callbacks=[FinetuningScheduler()], resume_from_checkpoint="some/path/to/my_checkpoint.ckpt")
```

Comment:
resume_from_checkpoint="some/path/to/my_checkpoint.ckpt" is currently depreceated. The new API is trainer.fit(..., ckpt_path=...)

Reply:
Good catch, forgot to update, thanks!